sdk/eventgrid/eventgrid/src/predicates.ts

```diff
@@ -440,11 +440,13 @@ export function isSystemEvent<T extends KnownSystemEventTypes>(
  * @param eventType The type of system event to check for, e.g., "Microsoft.AppConfiguration.KeyValueDeleted"
  * @param event The event to test.
  */
+// eslint-disable-next-line no-redeclare
```

Comment:
Seems like a bug. This is an overload signature and shouldn't count as a redeclaration.

Reply:
is it something silly like there shouldn't be an empty line between overload signatures?

Reply:
Okay I found a resolution: I used
```
    "no-redeclare": "off",
    "@typescript-eslint/no-redeclare": ["error", { builtinGlobals: true }],
```
instead of ESLint's vanilla `no-redeclare` and it did not exhibit this same bizarre behavior.

Reply:
Good catch. We should be using the typescript-awaire versions of all vanilla linting rules that have counterparts in typescript-eslint.

Reply:
Yeah I am looking into that now. And to make things weirder, we already had `"@typescript-eslint/no-redeclare": "off"` for some reason!